Fix activate widget tab no keep active after triggered save

mrtuvn opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->
Clean html class tabs (eliminate duplicate id for list item and anchor) and make sure tab work correctly like before
After this patch tab current keep active after admin change config and save form. Reload page return correct previous tab state
![Screenshot from 2021-02-16 21-41-44](https://user-images.githubusercontent.com/1908873/108078224-ff48bc80-709f-11eb-9493-900621ffd3d3.png)


Before after save form reload and set active for very first tab

### Related Pull Requests
<!-- related pull request placeholder -->

### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. Fixes magento/magento2#30166

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. Login admin
2. Go to Stores
3. Select Attributes in dropdown main menu
4. Go Edit Product attribute
5. Choose Storefront Properties tab (last tab)
6. Change some options in right content tab
7. Click` save and continue edit` and watch results

Expected results
When we selected tab ex: "Storefront Properties" and do some change in form. Then save. After page reload tab "Storefront" should keep active state instead

Actually results
The first tab active instead previous tab

### Questions or comments
<!---
	If relevant, here you can ask questions or provide comments on your pull request for the reviewer
	For example if you need assistance with writing tests or would like some feedback on one of your development ideas
-->
CC: @bvboas hope this fix correct as your wish
### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run Functional Tests CE, Functional Tests EE

mrtuvn commented:
@magento run Functional Tests CE

edenduong reviewed (changes requested):
Hi, 

Could you please cover this change by MFTF test?

Thanks !

mrtuvn commented:
Is there any existed test related with file changed or i need create new one

mrtuvn commented:
@magento give me test instance

mrtuvn commented:
@engcom-Charlie @AnnaAPak @engcom-Kilo Can you guys help with MFTF coverage
Area affect is attribute product edit page

mrtuvn commented:
CC @kate-kyzyma can you help with mftf coverage such case ?
This area in backend

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run Functional Tests B2B, Functional Tests EE, Static Tests

mrtuvn commented:
@magento run Functional Tests B2B, Functional Tests EE

mrtuvn commented:
@kate-kyzyma Hi can you help cover changes by mftf test

mrtuvn commented:
added mftf cover changes
@magento run all tests

mrtuvn commented:
@magento run Unit Tests, WebAPI Tests, Integration Tests, Functional Tests EE

edenduong commented on app/code/Magento/Catalog/Test/Mftf/Test/AdminVerifyStorefrontPropertiesTabStillActiveAfterSaveContinueEditTest.xml:
```diff
@@ -0,0 +1,65 @@
+
+        <!-- Go to Stores > Product, click "Add New Attribute" -->
+        <actionGroup ref="AdminOpenProductAttributePageActionGroup" stepKey="openProductAttributePage"/>
+        <click selector="{{AdminProductAttributeGridSection.createNewAttributeBtn}}" stepKey="createNewAttribute"/>
+        <waitForPageLoad stepKey="waitForNewAttributeFormLoad"/>
+
+        <!-- Verify properties storefront tab visible -->
+        <seeElement selector="{{StorefrontPropertiesSection.StoreFrontPropertiesTab}}" stepKey="seeTabAttributePropertiesVisible" />
```
I think you should create new ActionGroup

edenduong commented on app/code/Magento/Catalog/Test/Mftf/Test/AdminVerifyStorefrontPropertiesTabStillActiveAfterSaveContinueEditTest.xml:
```diff
@@ -0,0 +1,65 @@
+            <argument name="attributeType" value="{{productTextEditorAttribute.frontend_input}}"/>
+        </actionGroup>
+
+        <!-- Click on properties tab -->
+        <click selector="{{StorefrontPropertiesSection.StoreFrontPropertiesTab}}" stepKey="clickStorefrontPropertiesTab"/>
+
+        <!-- <scrollToTopOfPage stepKey="scrollToTabs"/> -->
+
+        <!-- Click on save continue -->
+        <click selector="{{AttributePropertiesSection.SaveAndEdit}}" stepKey="clickSave"/>
+
+        <!-- wait page reload done -->
+        <waitForPageLoad stepKey="waitForPageProductAttributeLoad"/>
+
+        <!-- Verify tab still active -->
+        <seeElement selector="//li[@id='list_product_attribute_tabs_front'][contains(@class, 'admin__page-nav-item ui-state-default ui-corner-top ui-tabs-active ui-state-active')]" stepKey="seeTabAttributePropertiesActive" />
```
Create new ActionGroup for assert the result

edenduong reviewed (changes requested):
Thank you for you contribution. Please see my comment.

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run Functional Tests CE, Functional Tests B2B, Functional Tests EE, Sample Data Tests B2B,  Sample Data Tests CE, Sample Data Tests EE

mrtuvn commented:
@magento run all tests

edenduong commented:
@magento run all tests

mrtuvn commented:
@magento run Functional Tests CE, Functional Tests B2B, Functional Tests EE

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run WebAPI Tests, Functional Tests CE, Functional Tests B2B

edenduong commented on app/code/Magento/Catalog/Test/Mftf/Test/AdminVerifyStorefrontPropertiesTabStillActiveAfterSaveContinueEditTest.xml:
```diff
@@ -0,0 +1,51 @@
+
+        <!-- Go to Stores > Product, click "Add New Attribute" -->
+        <actionGroup ref="AdminOpenProductAttributePageActionGroup" stepKey="openProductAttributePage"/>
+        <click selector="{{AdminProductAttributeGridSection.createNewAttributeBtn}}" stepKey="createNewAttribute"/>
+        <waitForPageLoad stepKey="waitForNewAttributeFormLoad"/>
```
COuld you please also move it to action group ? Tests should have only Action Group according to Magento MFTF Best Practice.

mrtuvn replied:
i have updated following mftf best practice

mrtuvn committed:
Fix activate widget tab no keep active after triggered save


update


update


up


update


update mftf file


update actiongroup


update


update action group

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run WebAPI Tests, Functional Tests B2B

magento-engcom-team commented:
Hi @edenduong, thank you for the review.
[ENGCOM-8976](https://jira.corp.magento.com/browse/ENGCOM-8976) has been created to process this Pull Request

engcom-Bravo commented:
Hi @mrtuvn,

We are closing this PR.The PR is in Draft since longtime.

Once you are ready Please feel free to reopen it to process the PR further.

Thanks
